.github/workflows/nomad.yaml

```diff
@@ -131,41 +130,81 @@ jobs:
 
           echo "alloc_ids=$alloc_ids" >> "$GITHUB_OUTPUT"
           echo "started_at=$(date +%s)" >> "$GITHUB_OUTPUT"
+          echo "job_name=${JOB_NAME}" >> "$GITHUB_OUTPUT"
 
       - name: Save alloc_ids to file
-        env:
-          RUN_ID: "${{ matrix.job-name || matrix.scenario-name }}_${{ github.run_id }}"
         run: |
           started_at=${{ steps.run-nomad-job.outputs.started_at }}
-          nix develop --command ./nomad/scripts/ci_allocs.sh make_allocs_csv alloc_ids.csv "${{ matrix.job-name || matrix.scenario-name }}" "${{ matrix.scenario-name }}" "${RUN_ID}" "${started_at}" "${{ steps.run-nomad-job.outputs.alloc_ids }}"
+          nix develop --command ./nomad/scripts/ci_allocs.sh make_allocs_csv alloc_ids.csv "${JOB_NAME}" "${{ matrix.scenario-name }}" "${RUN_ID}" "${started_at}" "${{ steps.run-nomad-job.outputs.alloc_ids }}"
 
       - name: Upload alloc_ids
         uses: actions/upload-artifact@v4
         with:
-          name: alloc_ids_${{ matrix.job-name || matrix.scenario-name }}
+          name: alloc_ids_${{ env.JOB_NAME }}
           path: alloc_ids.csv
 
-  wait-for-all-jobs:
-    name: Wait for all jobs to finish
-    runs-on: [self-hosted, wind-tunnel]
+      - name: Persist matrix job outputs for subsequent job use
+        uses: cloudposse/github-action-matrix-outputs-write@v1
+        with:
+          matrix-step-name: ${{ github.job }}
+          matrix-key: ${{ env.RUN_ID }}
+          outputs: |-
+            alloc_ids: ${{ steps.run-nomad-job.outputs.alloc_ids }}
+            run_id: ${{ env.RUN_ID }}
+            job_name: ${{ steps.run-nomad-job.outputs.job_name }}
+
+  collect-github-matrix-outputs:
```

Comment:
Why is this a separate job and not the last step in `run-scenarios`?

Reply:
This aggregates the results of the GITHUB_OUTPUT from each matrix job into a single data structure, so we can use it to define the next matrix.